cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -718,6 +718,14 @@ class ClientBuilder {
     if (!options_.region.empty()) {
       client_config_.region = ToAwsString(options_.region);
     }
+    if (options_.request_timeout_ms != -1) {
+      client_config_.requestTimeoutMs = std::lround(options_.request_timeout_ms);
+    }
+    if (options_.connect_timeout_ms != -1) {
+      client_config_.connectTimeoutMs = std::lround(options_.connect_timeout_ms);
+    }
```

Comment:
```suggestion
    if (options_.request_timeout_ms >= 0) {
      client_config_.requestTimeoutMs = std::lround(options_.request_timeout_ms);
    }
    if (options_.connect_timeout_ms >= 0) {
      client_config_.connectTimeoutMs = std::lround(options_.connect_timeout_ms);
    }
```

We should probably either do `>=0` (suggestion above) or return an error if it is less than `-1`